eva/executor/lateral_join_executor.py

```diff
@@ -27,24 +26,14 @@ def __init__(self, node: LateralJoinPlan):
         self.predicate = node.join_predicate
         self.join_project = node.join_project
 
-    def validate(self):
-        pass
-
     def exec(self, *args, **kwargs) -> Iterator[Batch]:
         outer = self.children[0]
         inner = self.children[1]
-        try:
-            for outer_batch in outer.exec(**kwargs):
-                for result_batch in inner.exec(lateral_input=outer_batch):
-                    result_batch = Batch.join(outer_batch, result_batch)
-                    result_batch.reset_index()
-                    result_batch = apply_predicate(result_batch, self.predicate)
-                    result_batch = apply_project(result_batch, self.join_project)
-                    if not result_batch.empty():
-                        yield result_batch
-        except Exception as e:
-            logger.error(e)
-            raise ExecutorError(e)
-
-    def __call__(self, *args, **kwargs) -> Generator[Batch, None, None]:
-        yield from self.exec(*args, **kwargs)
```

Comment:
Is ray supporting lateral join now? I think the `__call__` is needed for ray.

Reply:
Not sure about this \ cc @kaushikravichandran @jiashenC @gaurav274

Reply:
Even if it were the case, this code was not being used in any of the test cases and repeated in a few operators.

Reply:
Ideally, I think it could be triggered by Ray if there is a LateralJoin under Projection. Maybe we can move this to the AbstractExecutor to improve the coverage? I think it will anyway have the same function implementation for any executor.